portfolio/src/main/webapp/task.js

```diff
@@ -85,38 +87,117 @@ function updateTaskList(newTask, lengthUnit) {
   descriptionText.innerText = newTask.description;
   cardBody.appendChild(descriptionText);
 
-  const duration = document.createElement('p');
-  duration.classList.add('card-text');
-  duration.innerText = newTask.duration;
-  cardBody.appendChild(duration);
+  // Create row for all modifiable content
+  const inputRow = document.createElement('div');
+  inputRow.classList.add('form-row');
+
+  // Create a column for duration
+  const durationColumn = document.createElement('div');
+  durationColumn.classList.add('col');
+
+  const durationLabel = document.createElement('label');
+  durationLabel.setAttribute('for', 'duration-input-' + TASK_ID_COUNTER);
+  durationLabel.innerText = 'Duration:';
+
+  const durationInput = document.createElement('input');
+  durationInput.classList.add('form-control');
+  durationInput.setAttribute('id', 'duration-input-' + TASK_ID_COUNTER);
+  if (lengthUnit == TIME_UNIT.HOURS) {
+    durationInput.setAttribute('value', newTask.duration / 60);
+    $('#task-length-unit-message').removeClass('d-block');
+  } else if (lengthUnit == TIME_UNIT.MINUTES) {
+    durationInput.setAttribute('value', newTask.duration);
+    $('#task-length-unit-message').removeClass('d-block');
+  } else {
+    $('#task-length-unit-message').addClass('d-block');
+    return;
+  }
+
+  durationColumn.appendChild(durationLabel);
+  durationColumn.appendChild(durationInput);
+  inputRow.appendChild(durationColumn);
+
+  // Create a column for unit
+  const unitColumn = document.createElement('div');
+  unitColumn.classList.add('col');
+
+  const unitLabel = document.createElement('label');
+  unitLabel.setAttribute('for', 'unit-select-' + TASK_ID_COUNTER);
+  unitLabel.innerText = 'Unit:';
+
+  const unitSelect = document.createElement('select');
+  unitSelect.classList.add('form-control');
+  unitSelect.setAttribute('id', 'unit-select-' + TASK_ID_COUNTER);
+  unitSelect.setAttribute('selected', lengthUnit);
+
+  var optionMinutes = unitSelect.appendChild(document.createElement('option'));
```

Comment:
nit: I think `optionMinutes` and `optionHours` can be `const`?

Reply:
Done.